lang/llvm-12/files/0005-clang-support-macports-libstdcxx.patch

```diff
@@ -0,0 +1,202 @@
+                                                  arch == llvm::Triple::ppc64 ? "ppc64" : "");
+@@ -2103,10 +2112,17 @@ void DarwinClang::AddClangCXXStdlibIncludeArgs(
+ 
+     case llvm::Triple::x86:
+     case llvm::Triple::x86_64:
+-      IsBaseFound = AddGnuCPlusPlusIncludePaths(DriverArgs, CC1Args, UsrIncludeCxx,
+-                                                "4.2.1",
+-                                                "i686-apple-darwin10",
+-                                                arch == llvm::Triple::x86_64 ? "x86_64" : "");
++      IsBaseFound = false;
++      if (Type==ToolChain::CST_MacPortsLibstdcxx) {
++        IsBaseFound = AddGnuCPlusPlusIncludePaths(DriverArgs, CC1Args, llvm::StringRef("@@MACPORTS_GCC_INCLUDE_DIR@@"),
++                                                  "",
++                                                  "@@MACPORTS_HOST_NAME@@",
++                                                  arch == llvm::Triple::x86_64 ? "@@MACPORTS_GCC_SUBDIRECTORY_x86_64@@" : "@@MACPORTS_GCC_SUBDIRECTORY_i386@@");
++      }
++      IsBaseFound |= AddGnuCPlusPlusIncludePaths(DriverArgs, CC1Args, UsrIncludeCxx,
++                                                 "4.2.1",
++                                                 "i686-apple-darwin10",
++                                                 arch == llvm::Triple::x86_64 ? "x86_64" : "");
+       IsBaseFound |= AddGnuCPlusPlusIncludePaths(DriverArgs, CC1Args, UsrIncludeCxx,
+                                                 "4.0.0", "i686-apple-darwin8",
+                                                  "");
+@@ -2149,6 +2165,12 @@ void DarwinClang::AddCXXStdlibLibArgs(const ArgList &Args,
+     break;
+ 
+   case ToolChain::CST_Libstdcxx:
++  case ToolChain::CST_MacPortsLibstdcxx:
++    if (Type==ToolChain::CST_MacPortsLibstdcxx && getVFS().exists("@@MACPORTS_libstdc++@@")) {
++      CmdArgs.push_back("@@MACPORTS_libstdc++@@");
++      return;
++      }
++
+     // Unfortunately, -lstdc++ doesn't always exist in the standard search path;
+     // it was previously found in the gcc lib dir. However, for all the Darwin
+     // platforms we care about it was -lstdc++.6, so we search for that
+diff --git a/clang/lib/Driver/ToolChains/Hexagon.cpp b/clang/lib/Driver/ToolChains/Hexagon.cpp
+index fb54f73bcd4c..0260a32096af 100644
+--- a/clang/lib/Driver/ToolChains/Hexagon.cpp
++++ b/clang/lib/Driver/ToolChains/Hexagon.cpp
+@@ -643,6 +643,8 @@ HexagonToolChain::GetCXXStdlibType(const ArgList &Args) const {
+       return ToolChain::CST_Libstdcxx;
+   }
+   StringRef Value = A->getValue();
++  if (Value == "macports-libstdc++")
++    return ToolChain::CST_MacPortsLibstdcxx;
+   if (Value != "libstdc++" && Value != "libc++")
+     getDriver().Diag(diag::err_drv_invalid_stdlib_name) << A->getAsString(Args);
+ 
+diff --git a/clang/lib/Frontend/CompilerInvocation.cpp.rej b/clang/lib/Frontend/CompilerInvocation.cpp.rej
```

Comment:
yoops. this bit snuck in and is not needed. Causes no harm, but does no good. I'll get rid of this.